mm2src/coins/eth/eth_swap_v2/eth_maker_swap_v2.rs

```diff
@@ -0,0 +1,509 @@
+                maker_secret_hash: try_tx_s!(args.maker_secret_hash.try_into()),
+                payment_time_lock: args.time_lock,
+            }
+        };
+        match &self.coin_type {
+            EthCoinType::Eth => {
+                let data = try_tx_s!(self.prepare_maker_eth_payment_data(&payment_args).await);
+                self.sign_and_send_transaction(
+                    payment_amount,
+                    Action::Call(maker_swap_v2_contract),
+                    data,
+                    U256::from(self.gas_limit_v2.maker.eth_payment),
+                )
+                .compat()
+                .await
+            },
+            EthCoinType::Erc20 {
+                platform: _,
+                token_addr,
+            } => {
+                let data = try_tx_s!(
+                    self.prepare_maker_erc20_payment_data(&payment_args, payment_amount, *token_addr)
+                        .await
+                );
+                self.handle_allowance(maker_swap_v2_contract, payment_amount, args.time_lock)
+                    .await?;
+                self.sign_and_send_transaction(
+                    U256::from(ZERO_VALUE),
+                    Action::Call(maker_swap_v2_contract),
+                    data,
+                    U256::from(self.gas_limit_v2.maker.erc20_payment),
+                )
+                .compat()
+                .await
+            },
+            EthCoinType::Nft { .. } => Err(TransactionErr::ProtocolNotSupported(ERRL!(
+                "NFT protocol is not supported for ETH and ERC20 Swaps"
+            ))),
+        }
+    }
+
+    pub(crate) async fn validate_maker_payment_v2_impl(
+        &self,
+        args: ValidateMakerPaymentArgs<'_, Self>,
+    ) -> ValidatePaymentResult<()> {
+        if let EthCoinType::Nft { .. } = self.coin_type {
+            return MmError::err(ValidatePaymentError::ProtocolNotSupported(
+                "NFT protocol is not supported for ETH and ERC20 Swaps".to_string(),
+            ));
+        }
+        let maker_swap_v2_contract = self
+            .swap_v2_contracts
+            .as_ref()
+            .map(|contracts| contracts.maker_swap_v2_contract)
+            .ok_or_else(|| {
+                ValidatePaymentError::InternalError("Expected swap_v2_contracts to be Some, but found None".to_string())
+            })?;
+        validate_payment_args(args.taker_secret_hash, args.maker_secret_hash, &args.amount)
+            .map_to_mm(ValidatePaymentError::InternalError)?;
+        let maker_address = public_to_address(args.maker_pub);
+        let swap_id = self.etomic_swap_id_v2(args.time_lock, args.maker_secret_hash);
+        let maker_status = self
+            .payment_status_v2(
+                maker_swap_v2_contract,
+                Token::FixedBytes(swap_id.clone()),
+                &MAKER_SWAP_V2,
+                EthPaymentType::MakerPayments,
+                2,
```

Comment:
Can we create consts for those to for readability?

Reply:
Agree, had feeling that smth should be done here, but couldnt understand what
Done https://github.com/KomodoPlatform/komodo-defi-framework/pull/2211/commits/2b81855af3a52121d924eade11c1abf423e074ee